pkg/cli/demo.go

```diff
@@ -164,6 +165,13 @@ func setupTransientServers(
 		if i == 0 {
 			s = serv
 		}
+
+		// Start up the update check loop.
+		// We don't do this in (*server.Server).Start() because we don't want it
+		// in tests.
+		if !envutil.EnvOrDefaultBool("COCKROACH_SKIP_UPDATE_CHECK", false) {
```

Comment:
I'd recommend/request capturing the entire conditional into a common function (`maybeEnableTelemetry`) so that we don't the run the risk of diverging the mechanisms over time.